src/services/UserService.js

```diff
@@ -23,8 +41,54 @@ const UserService = (function () {
     }
 
     function saveUserSettingsToLS () {
-        for (let k in userSettings) {
-            localStorage.setItem(`user-${k}`, userSettings[k]);
+        const userToString = JSON.stringify(userSettings);
+        localStorage.setItem('user', userToString);
+    }
+
+    function getUserObjFromLS () {
+        return new Promise((resolve, reject) => {
+            setTimeout(() => {
+                try {
+                    const userFromLS = localStorage.getItem('user');
+                    resolve(userFromLS);
+                } catch (error) {
+                    reject(error);
+                }
+            }, 300);
+        });
+    }
+
+    function saveLiquidConsumption (date, water) {
+        return {
+            date,
+            water
+        }
+    }
+
+    function updateConsumptionArr (item) {
+        // userSettings.consumption.push(item);
+
+        // WIP
+        const consumptionArr = userSettings.consumption;
+        const today = getCurrentDay();
+
+        if (consumptionArr.length === 0) {
```

Comment:
I don't follow this, but maybe you need something like this:

```suggestion
       const todayItemIndex = consumptionArr.findIndex((i) => i.date === today);
       if (todayItemIndex === -1) {
            consumptionArr.push(item);
        } else {
           consumptionArr[todayItemIndex].water += 200;
        }
```